### ISSUE TITLE
Communication problems between two dds demo running on two computers

### ISSUE THREAD

**Issue by aohwang** [eclipse-cyclonedds/cyclonedds#1699]
Hi friends,

I have using CycloneDDS in several projects, and it behaves excellently.  In the recent days, I tried release version 0.10.2 in my testing environment, but I couldn't make it work. 

There are two nodes in my testing environment, including one Win 10 desktop (192.168.4.23) and embedded development board (192.168.4.127).

The testing programs are simple, the program running on Win 10 desktop is sending samples repeatedly and the program running on the embedded development board is just receiving messages.

I enabled the finest log level in the two programs and captured the network traffic using wireshark. After I walk around the logs and traffic, I found some strange behaviours.

The program running on node 192.168.4.127 is tring to write sedp messages, however no packets are sent to the network. You can see it from the picture.
![image](https://github.com/eclipse-cyclonedds/cyclonedds/assets/43724668/8d45e12d-a731-4383-aa60-cf695c5b363f)

Most of the packets captured are from 192.168.4.23 to 192.168.4.127. The packets sent from 192.168.4.127 are just SPDP packets and some Heartbeat and ACKNACK messages.

Could you give me some suggestions about how to fix the problem? Anything would be appreciated. 

Please remove the ".txt" suffix of the rtps_capture.pcapng.txt and open it with wireshark. 

[node1_192.168.4.127.txt](https://github.com/eclipse-cyclonedds/cyclonedds/files/11543001/node1_192.168.4.127.txt)
[node2_192.168.4.23.txt](https://github.com/eclipse-cyclonedds/cyclonedds/files/11543003/node2_192.168.4.23.txt)
[rtps_capture.pcapng.txt](https://github.com/eclipse-cyclonedds/cyclonedds/files/11543011/rtps_capture.pcapng.txt)

**Comment by quantking001** [comment]
Thanks @eboasson , I have solved the problem. 

The first odd thing you mentioned is about the time inconsistency. It's because the time synchronization in the embedded development board is not working. 

You remind me that the `tev` thread may be not able to keep up. The operating system running on the board is VxWorks. I checked the CPU usage and found that it's fully occupied by one task, so the `tev` thread gets no chance to run. After I stopped the task, all things worked. Thanks for your detailed response. :)
